sentry/src/main/java/io/sentry/CombinedScopeView.java

```diff
@@ -454,7 +455,6 @@ public void bindClient(@NotNull ISentryClient client) {
 
   @Override
   public @NotNull ISentryClient getClient() {
-    // TODO [HSM] checking for noop here doesn't allow disabling via client, is that ok?
```

Comment:
Discussed internally with backend team, if a customer needs this, they can provide their own noop `ISentryClient` that isn't actually `NoOpSentryClient` and thus not detected here.